textproc/scdoc/files/patch-makefile.diff

```diff
@@ -1,7 +1,7 @@
 --- Makefile.orig       2019-03-04 16:43:31.000000000 +0100
 +++ Makefile    2019-06-05 16:54:49.000000000 +0200
 @@ -1,6 +1,5 @@
- VERSION=1.9.4
+ VERSION=1.10.1
```

Comment:
The usual way to handle situations like this is to change it to a variable name in the patch and then set the variable to the desired value in the post-patch phase in the portfile. That way the patch does not need to be updated every version update. 

See also https://trac.macports.org/wiki/PortfileRecipes#hardcode_opt_local

Reply:
But is it possible to reinplace patch file before applying it? post-patch phase is too late for it. As far as in such case we should reinplace patch file, not a resulting Makefile.

Reply:
sure, why do you think the `post-patch` phase is too late? You can change your patchfile to result in, for example, `VERSION=@@VERSION@@` and then do the reinplace to get the `@@VERSION@@` variable replaced by the actual number. 
```
post-patch {
    reinplace "s|@@VERSION@@|${version}|g" ${worksrcpath}/Makefile}
```

Reply:
Well, if I change patchfile, it will include version anyway:
```diff
-VERSION=1.10.1
+VERSION=@@VERSION@@
```
So, there is no profit for us.
Or we should reinplace patchfile before applying patch, something like:
```
pre-patch {
    reinplace "s|@@VERSION@@|${version}|g" files/patch-makefile.diff
```
post-patch phase is definitely too late for it.

Reply:
Sorry to jump into y'alls PR discussion, but I thought I'd share my portfile for the latest scdoc. I've pushed it up but haven't created a PR because this one already exists.

I removed the patch file, removed the reinplace for the prefix in the Makefile, but added one to replace the LD flag attempting to-do a static link.

[Here are the changes I made but finally committed and pushed up.](https://github.com/chadcatlett/macports-ports/commit/eeb12d072d7850ab3a4bfd4d6c09917391215957)

I only link it as an example of another approach.

Reply:
That's definitely a better approach. Are you planning to make PR yourself or I can adjust current PR with your approach?

Reply:
@sikmir either way is fine to me. I didn't submit a PR because as I was filing out the checklist I searched for an update to scdoc and found this PR.

Reply:
@sikmir I just thought of something. I'd probably change the tests from 

```
test.run            yes
test.cmd            make check
```
to something along the lines of
```
test.run            yes
test.target         check
```

Reply:
Updated.

Reply:
@mf2k Is it OK for you? Could we land this PR?